src/Microsoft.AspNetCore.Razor.Language/DefaultRazorIRLoweringPhase.cs

```diff
@@ -134,12 +134,19 @@ public override void VisitImportSpan(AddImportChunkGenerator chunkGenerator, Spa
 
             public override void VisitAddTagHelperSpan(AddTagHelperChunkGenerator chunkGenerator, Span span)
             {
-                _builder.Push(new DirectiveIRNode()
+                var directiveNode = new DirectiveIRNode()
                 {
                     Name = CSharpCodeParser.AddTagHelperDirectiveDescriptor.Directive,
                     Descriptor = CSharpCodeParser.AddTagHelperDirectiveDescriptor,
                     Source = BuildSourceSpanFromNode(span),
-                });
+                };
+
+                for (var i = 0; i < chunkGenerator.Diagnostics.Count; i++)
+                {
+                    directiveNode.Diagnostics.Add(chunkGenerator.Diagnostics[i]);
+                }
+
+                _builder.Push(directiveNode);
```

Comment:
>Malformed extensible directives. 1 lowering point.....Malformed built-in directives. These each have their own lowering points.

None of these are arguments about why this is the best design, they are arguments for why this is the least code required to implement the design.